client/modules/IDE/components/ConsoleInput.jsx

```diff
@@ -23,76 +24,99 @@ function ConsoleInput({ theme, dispatchConsoleEvent, fontSize }) {
       mode: 'javascript',
       inputStyle: 'contenteditable'
     });
+  }, []);
 
-    cmInstance.current.on('keydown', (cm, e) => {
+  useEffect(() => {
+    const handleEnterKey = (cm, e) => {
       if (e.key === 'Enter' && !e.shiftKey) {
         e.preventDefault();
         e.stopPropagation();
-        const value = cm.getValue();
-        if (value.trim(' ') === '') {
-          return false;
-        }
+
+        const value = cm.getValue().trim();
+        if (value === '') return;
+
         const messages = [
           { log: Encode({ method: 'command', data: [value] }) }
         ];
         const consoleEvent = [{ method: 'command', data: [value] }];
+
         dispatchMessage({
           type: MessageTypes.EXECUTE,
-          payload: {
-            source: 'console',
-            messages
-          }
+          payload: { source: 'console', messages }
         });
+
         dispatchConsoleEvent(consoleEvent);
```

Comment:
Wouldn't you need to wrap this in `dispatch = useDispatch()`, `dispatch(...)`, like the reducers used in `Console.jsx`? Asking since, currently, while Up arrow does work, the console no longer echoes the inputted commands. (:

Reply:
Yes!! Thank you so much for catching that 😂 I thought I had it in at some point! I'll try to have the update out for this at some point early next week.

Reply:
The update works, thanks a ton! :tada: